src/main/src/main/java/org/geoserver/security/impl/DefaultResourceAccessManager.java

```diff
@@ -446,7 +446,7 @@ public WorkspaceAccessLimits getAccessLimits(Authentication user, WorkspaceInfo
 
         CatalogMode mode = getMode();
 
-        if (readable && writable) {
+        if (readable && writable && !adminable) {
```

Comment:
I'm confused by this change... looking at the ticket, we are dealing with a workspace administrator.
The code authorizing the components visibility should be the [WorkspaceAdminComponentAuthorizer](https://github.com/geoserver/geoserver/blob/6bb431a36a57aa1b6cf68d78bc7dea35b0958812/src/web/core/src/main/java/org/geoserver/web/WorkspaceAdminComponentAuthorizer.java#L52) which should allow visibility when there is at least one adminable workspace. Are you sure that, with this change, links are not getting visible also for users that cannot admin anything?

Reply:
I tested the change for users who have no admin privilege at all, and links are not visible for them. It seems that this change solves the issue without side effects. 

Anyway, I changed the if statement organization, to increase its readability.

Reply:
@aaime In order to make sure that REST API works well, I have tested access to the REST API page after this change. Normal users are denied access to REST API, whereas admins are allowed.